File: OPENAPI-REST-API/swagger-client/dart/lib/model/license_order_request.dart
part of swagger.api;

class LicenseOrderRequest {
  /* License service type id from getNewLicense.serviceTypes (must be buyable). */
  int package = null;
/* IP address the license is bound to. */
  String ip = null;
/* Terms-of-service acceptance. Must be true to place the order. */
  bool tos = null;
/* Billing frequency in months. */
  int frequency = 1;
/* Coupon code. */
  String coupon = "";
/* Free-form note saved on the service row. */
  String comment = "";

  LicenseOrderRequest();

  @override
  String toString() {
    return 'LicenseOrderRequest[package=$package, ip=$ip, tos=$tos, frequency=$frequency, coupon=$coupon, comment=$comment, ]';
  }

  LicenseOrderRequest.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    package = json['package'];
    ip = json['ip'];
    tos = json['tos'];
    frequency = json['frequency'];
    coupon = json['coupon'];
    comment = json['comment'];
  }

  Map<String, dynamic> toJson() {
    return {
      'package': package,
      'ip': ip,
      'tos': tos,
      'frequency': frequency,
      'coupon': coupon,
      'comment': comment
     };
  }

  static List<LicenseOrderRequest> listFromJson(List<dynamic> json) {
    return json == null ? new List<LicenseOrderRequest>() : json.map((value) => new LicenseOrderRequest.fromJson(value)).toList();
  }

  static Map<String, LicenseOrderRequest> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, LicenseOrderRequest>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new LicenseOrderRequest.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/LicenseOrderRequest.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Request body to validate (putLicenses) or place (addLicense) a license order. Package ids come from getNewLicense.serviceTypes.
  /// </summary>
  [DataContract]
  public class LicenseOrderRequest {
    /// <summary>
    /// License service type id from getNewLicense.serviceTypes (must be buyable).
    /// </summary>
    /// <value>License service type id from getNewLicense.serviceTypes (must be buyable).</value>
    [DataMember(Name="package", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "package")]
    public int? Package { get; set; }

    /// <summary>
    /// IP address the license is bound to.
    /// </summary>
    /// <value>IP address the license is bound to.</value>
    [DataMember(Name="ip", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ip")]
    public string Ip { get; set; }

    /// <summary>
    /// Terms-of-service acceptance. Must be true to place the order.
    /// </summary>
    /// <value>Terms-of-service acceptance. Must be true to place the order.</value>
    [DataMember(Name="tos", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tos")]
    public bool? Tos { get; set; }

    /// <summary>
    /// Billing frequency in months.
    /// </summary>
    /// <value>Billing frequency in months.</value>
    [DataMember(Name="frequency", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "frequency")]
    public int? Frequency { get; set; }

    /// <summary>
    /// Coupon code.
    /// </summary>
    /// <value>Coupon code.</value>
    [DataMember(Name="coupon", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "coupon")]
    public string Coupon { get; set; }

    /// <summary>
    /// Free-form note saved on the service row.
    /// </summary>
    /// <value>Free-form note saved on the service row.</value>
    [DataMember(Name="comment", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "comment")]
    public string Comment { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class LicenseOrderRequest {\n");
      sb.Append("  Package: ").Append(Package).Append("\n");
      sb.Append("  Ip: ").Append(Ip).Append("\n");
      sb.Append("  Tos: ").Append(Tos).Append("\n");
      sb.Append("  Frequency: ").Append(Frequency).Append("\n");
      sb.Append("  Coupon: ").Append(Coupon).Append("\n");
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
